File: leetcode/easy/1389/1389.cpp
/**
 * Runtime: 0 ms, faster than 100.00%
 * Memory Usage: 8.8 MB, less than 34.60%
 */
class Solution {
  public:
    vector<int> createTargetArray(vector<int>& nums, vector<int>& index) {
      vector<int> result;

      for (int i=0; i<index.size(); ++i)
      {
        auto it = result.begin();
        result.insert(it + index[i], nums[i]);
      }


      return result;
    }
};

File: leetcode/easy/1389/1389.rb
# Runtime: 40 ms, faster than 90.00% 
# Memory Usage: 210 MB, less than 10.00% 
def create_target_array(nums, index)
  result = []
  for i in (0...index.size) do
    result.insert(index[i], nums[i]) 
  end

  result
end
